src/PersistentStorage.cpp

```diff
@@ -38,7 +56,40 @@ std::string PersistentStorage::get(const std::string& key, bool* found) {
     return value;
 }
 
+std::string PersistentStorage::getRange(const std::string& key_from, const std::string& key_to) {
+    std::string key = "";
+    rocksdb::Iterator* it = _db->NewIterator(rocksdb::ReadOptions());
+  
+    Document result;
+    result.SetObject();
+
+    for (it->SeekToFirst(); it->Valid(); it->Next()) {
+        key = it->key().ToString();
+        if((strcmp(key.c_str(), key_from.c_str()) >= 0) && (strcmp(key.c_str(), key_to.c_str()) <= 0))
```

Comment:
There is a much more efficient way to do a range query using rocksdb API without iterating the entire database.

Reply:
Merged with minor changes https://github.com/afiskon/hurmadb/commit/755987a95646892beab25f0b0ea3cdea3d86ff81